<?xml version="1.0"?>
<launch>
  <arg name="arm_controller_type" default="velocity" />
  <param name="arm_controller_type" value="$(arg arm_controller_type)" />
  <group ns="bringup_servo_node">
    <param name="publish_joint_positions" value="false" />
    <param name="publish_joint_velocities" value="true" />
    <rosparam command="load" file="$(find stir_ros)/config/moveit_servo.yaml" />
    <rosparam command="load" file="$(find stir_ros)/config/pose_tracking_settings.yaml" />
    <group if="$(eval arg('arm_controller_type') == 'velocity')">
      <param name="controller_command_topic" value="/crane_x7/arm_velocity_controller/command" />
      <param name="command_out_topic" value="/moveit_servo_command" />
      <param name="command_out_type" value="std_msgs/Float64MultiArray" />
      <node name="command_order_sorter" pkg="stir_ros" type="command_order_sorter.py" output="screen" />
    </group>
  </group>
  <node name="bringup_servo_node" pkg="stir_ros" type="bringup_servo_node" output="screen" />
</launch>
